services/blockstorage/test/block_sync_server_test.go

```diff
@@ -0,0 +1,129 @@
+package test
+
+import (
+	"context"
+	"errors"
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/crypto/keys"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestSourceRespondToAvailabilityRequests(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		harness := newHarness(ctx)
+		harness.setupSomeBlocks(3)
+		harness.gossip.When("SendBlockAvailabilityResponse", mock.Any).Return(nil, nil).Times(1)
+		msg := builders.BlockAvailabilityRequestInput().
+			WithFirstBlockHeight(1).
+			WithLastCommittedBlockHeight(primitives.BlockHeight(2)).
+			WithLastBlockHeight(primitives.BlockHeight(2)).
+			Build()
+		_, err := harness.blockStorage.HandleBlockAvailabilityRequest(msg)
+
+		require.NoError(t, err, "expecting a happy flow")
+
+		harness.verifyMocks(t, 1)
+	})
+}
+
+func TestSourceRespondsNothingToAvailabilityRequestIfSourceIsBehindPetitioner(t *testing.T) {
```

Comment:
is the name appropriate? looks like the behavior you expect is `DoesNotRespondToAvailabilityRequest`

Reply:
correct